sumologic/provider.go

```diff
@@ -36,6 +36,11 @@ func Provider() terraform.ResourceProvider {
 				Optional: true,
 				Default:  os.Getenv("SUMOLOGIC_BASE_URL"),
 			},
+			"admin": {
```

Comment:
I don't think is the right place for setting adminMode. The admin mode is only applicable to certain resources and should not be a global property. Besides having adminMode here will turn it on or off for all the resources. We want to toggle admin mode for each resource rather than globally.

If the goal is to get rid of adminMode and simplify call sites, we should look into some other option. Maybe have 2 versions of Get/Post/Put methods like GetInAdminMode/PostInAdminMode.

Reply:
The pattern I followed here is similar to configuring an AWS provider to spin up resources in a different region; create a second provider and alias it. The way this would work in practice with sumologic would be:

```
provider "sumologic" {
  alias    = "admin"
  admin    = true
  base_url = "https://long-api.sumologic.net/api/"
}
```

Which would only be used in corresponding resources that need/want the admin call:

```# data "sumologic_admin_recommended_folder" "admin" {}

resource "sumologic_folder" "folder" {
  provider    = sumologic.admin
  name        = "test-folder"
  description = "A test folder"
  parent_id   = data.sumologic_admin_recommended_folder.admin.id
}
```

Reply:
Ah, I see, I wasn't aware of this pattern. Yeah, this would work. However, this will a breaking change, as earlier we used to default to true for the admin recommended folder.

Reply:
We should call this field "adminMode".

Reply:
Is there a test that should pass/fail with this change? They pass locally for me. Additionally, the data source for `sumologic_admin_recommended_folder` seems to still work as well. The API docs don't require the header to get the folder ID: https://api.sumologic.com/docs/#operation/getAdminRecommendedFolderAsync ; it's only required when creating folders underneath it.

Reply:
>The API docs don't require the header to get the folder ID: api.sumologic.com/docs/#operation/getAdminRecommendedFolderAsync ; it's only required when creating folders underneath it.

I thought we did, but looks like we don't. This will not be a breaking change in that case. 

>Is there a test that should pass/fail with this change?

There is https://github.com/SumoLogic/terraform-provider-sumologic/blob/master/sumologic/data_source_sumologic_admin_recommended_folder_test.go